src/generator/models.ts

```diff
@@ -532,3 +548,94 @@ function generateAliasType(structDef: StructDef, receiver: string, forMarshal: b
   text += `\t}\n`;
   return text;
 }
+
+function generateAdditionalPropertiesMarshaller(structDef: StructDef, parentType?: ObjectSchema) {
+  imports.add('encoding/json');
+  const typeName = structDef.Language.name;
+  const receiver = typeName[0].toLowerCase();
+  // generate marshaller method
+  let marshaller = `func (${receiver} ${typeName}) MarshalJSON() ([]byte, error) {\n`;
+  marshaller += '\tobjectMap := make(map[string]interface{})\n';
+  const emitMarshaller = function (prop: Property): string {
+    let text = `\tif ${receiver}.${prop.language.go!.name} != nil {\n`;
+    if (prop.schema.language.go!.internalTimeType) {
+      text += `\t\tobjectMap["${prop.serializedName}"] = (*${prop.schema.language.go!.internalTimeType})(${receiver}.${prop.language.go!.name})\n`;
+    } else {
+      text += `\t\tobjectMap["${prop.serializedName}"] = ${receiver}.${prop.language.go!.name}\n`;
+    }
+    text += `\t}\n`;
+    return text;
+  }
+  // TODO: multiple inheritance
+  for (const prop of values(parentType?.properties)) {
+    if (prop.language.go!.isAdditionalProperties) {
+      continue;
+    }
+    marshaller += emitMarshaller(prop);
+  }
+  for (const prop of values(structDef.Properties)) {
+    if (prop.language.go!.isAdditionalProperties) {
+      continue;
+    }
+    marshaller += emitMarshaller(prop);
+  }
+  marshaller += `\tif ${receiver}.AdditionalProperties != nil {\n`;
```

Comment:
Just checking, will additional properties always be in a field called AdditionalProperties?

Reply:
Yep!  This is what we did for track 1 and is what the other languages do as well.